code/modules/mob/living/living.dm

```diff
@@ -459,6 +472,31 @@
 	update_mobility()
 
 
+/mob/living/proc/get_up()
+	set waitfor = FALSE
+	var/static/datum/callback/rest_checks = CALLBACK(src, .proc/rest_checks_callback)
+	if(!do_mob(src, src, 2 SECONDS, uninterruptible = TRUE, extra_checks = rest_checks))
+		return
+	if(resting || lying_angle == 0 || HAS_TRAIT(src, TRAIT_FLOORED))
+		return
+	set_lying_angle(0)
+
+
+/mob/living/proc/rest_checks_callback()
+	if(resting || lying_angle == 0 || HAS_TRAIT(src, TRAIT_FLOORED))
+		return FALSE
+	return TRUE
+
+
+/mob/living/proc/set_lying_down(new_lying_angle)
+	if(buckled && buckled.buckle_lying == 0)
```

Comment:
I really hate to ask, but why not use ? here?

Reply:
I've seen cases where comparisons fail with `?`
Might be cargo culting at this point, but I feel this to be more clear.